query-engine/core/src/telemetry/capturing/mod.rs

```diff
@@ -158,9 +158,13 @@ pub fn capturer(trace_id: trace::TraceId, settings: Settings) -> Capturer {
 
 /// Adds a capturing layer to the given subscriber and installs the transformed subscriber as the
 /// global, default subscriber
+#[cfg(feature = "metrics")]
 #[allow(clippy::type_complexity)]
 pub fn install_capturing_layer(
-    subscriber: Layered<Option<MetricRegistry>, Layered<Box<dyn Layer<Registry> + Send + Sync>, Registry>>,
+    subscriber: Layered<
+        Option<query_engine_metrics::MetricRegistry>,
+        Layered<Box<dyn Layer<Registry> + Send + Sync>, Registry>,
+    >,
```

Comment:
Background info: this is used by the binary engine only, so we should place it in a different crate, and avoid importing it in the napi/wasm libquery case.